rmgpy/data/kinetics/family.py

```diff
@@ -1795,14 +1804,27 @@ def __generateReactions(self, reactants, products=None, forward=True, prod_reson
         else:
             template = self.reverseTemplate
 
+        if len(reactants) > len(template.reactants): #if the family has one template and is unimolecular split template into multiple reactants
```

Comment:
Ok, so this has been bothering me, and I would additional confirmation that it won't have any adverse effects. The motivating concern was why the deepcopy issue only affected the liquid oxidation test. So I analyzed this section again, and have a few more thoughts:
1. The comment says `if the family has one template and is unimolecular`, which doesn't seem to match with the if condition. Should it be `if the family has one template and is multimolecular`? This is probably just a typo though, so we don't need to dwell on it.
2. So then when do we trigger this if clause? The obvious case is for the AG-trees where all families have a single template. However, it can also happen if we call `__generateReactions` in the forward direction with the products of a family, e.g. 2 products for the unimolecular `HO2_Elimination_from_PeroxyRadical` family or 3 products for the `Peroxyl_Disproportionation` family. Both of these families (and other trimolecular families) are liquid phase families and are only included in the liquid oxidation example. Of course, the same thing would happen when calling `__generateReactions` in the reverse direction with the reactants of a family, e.g. 2 reactants for `R_Addition_MultipleBond`, which leaves me wondering about why we didn't see more of performance impact on the other tests.
3. Thus, it seems like this might trigger in more situations than originally expected (at least for me, idk if you considered those other cases). However, I don't think any of the existing families have disconnected graphs, so split should return the original graph, and we end up with the same template entries.
4. Thinking about the `deepcopy` again, you're creating multiple entries and assigning the split groups to each entry. However, without the deepcopy, it seems like each entry would point to the same instance, so you're simply replacing the groups of all them each time. This doesn't have any apparent effects because there aren't any groups that can be split currently, but it could have effects for atg4.

Reply:
1.  Yes
2.  Does forward=True reaction generation generate reverse reactions?  My code is based on the assumption that I know if it's being reacted in the forward or reverse direction and use the forward/reverse templates as appropriate to make it so this should only happen to trees that have a single template that represents more than one reactant.  My thought was that this wasn't the case and that one of the liquid families might be written more like an atg family?
3.  Yes
4.  Absolutely true, fortunately It appears I've already fixed this in atg4 in a way that doesn't require a deepcopy by restructuring what objects are used.  see:  https://github.com/ReactionMechanismGenerator/RMG-Py/commit/e59fcb6c03243072cd79175593c8586fd309265f

My thought is that it should be safe to use this branch now especially since this branch can't run atg trees in RMG jobs yet.  I'd like to resolve 2 so that I can fix as needed on atg4, but I think the more fundamental issue is solved on atg4 already.

Reply:
It should be fine for now, but it's actually not very smart to split the templates everytime the function is called.  The templates should probably be generated in the init method of the family so they don't need to be generated each function call.

Reply:
Since we try generating reactions in both directions for a given set of reacting species, it should be quite common to encounter a mismatch in the number of species and templates. I was also thinking that maybe one of the liquid families was written like an atg family, but I didn't find any.